tests/student_page_test.py

```diff
@@ -147,4 +158,39 @@ def test_error_message_displayed(page: Page):
 
     # Use expect to wait until the error message appears
     error_locator = page.locator("#code-output")
+    expect(error_locator).to_contain_text("Error:", timeout=10000)
     expect(error_locator).to_contain_text("Error:", timeout=20000)
+
+
+def test_timer(page: Page):
+    """
+    Test that the timer runs at the start of the page and stops
+    when the correct output is found
+    """
+
+    # 1. Check timer 1 after 1 second
+    # time.sleep(1)
+    # Test suit seems to take 1 second before it gets here
+    expect(page.locator("#timer_val")).to_have_text("00:00:01")
+
+    # Check timer after 13 seconds
+    time.sleep(12)
+    expect(page.locator("#timer_val")).to_have_text("00:00:13")
+
+    # Check timer after 1 minute
+    time.sleep(47)
```

Comment:
It might make sense to use [clock](https://playwright.dev/python/docs/clock) to speed up this process.

Reply:
Since you aren't feeling well, I went ahead and pushed a potential change for this to your branch. This seemed simpler than trying to explain what I was thinking.